src/drive/web/modules/move/MoveModal.jsx

```diff
@@ -89,10 +97,39 @@ class MoveModal extends React.Component {
     }
   }
 
-  moveEntry = (entryId, destinationId) => {
-    return this.context.client
-      .collection('io.cozy.files')
-      .updateFileMetadata(entryId, { dir_id: destinationId })
+  /**
+   * Try to move entry to destination.
+   *
+   * @param   {string} entryId       - The file's id
+   * @param   {string} destinationId - The target directory's id
+   * @param   {string} filename      - The file's name
+   * @returns {Promise}              - A promise that returns the move action response if resolved or an Error if rejected
+   *
+   */
+  moveEntry = async (entryId, destinationId, filename) => {
+    const { client } = this.context
+    try {
+      return await client
+        .collection('io.cozy.files')
+        .updateFileMetadata(entryId, { dir_id: destinationId })
+    } catch (e) {
+      if (flag('handle-move-conflicts') && e.status === 409) {
+        const targetPath = await CozyFile.getFullpath(destinationId, filename)
+        if (this.props.sharingState.sharedPaths.includes(targetPath)) {
```

Comment:
Is there a way to not having to rely on props.sharingState here? And having a new arg or something?

Reply:
If we want to move this method out of here, we could pass `sharedPaths` as an argument (not really pretty). Or maybe we could pass a `overwriteIfConflict` argument but in this case, we have to move the path calculation and flag check out of the function. I'm OK with this.

Reply:
I read the code again. I think it is coherent to rely on props since the component is coupled to the sharingProvider, BUT I think `moveEntry` should not be part of the component but more part of the "model". 

How will you know if the file is shared with a `overwriteIfConflict` argument? You will check if the file is shared before calling `moveEntry` ?